kedro/framework/cli/starters.py

```diff
@@ -631,21 +635,46 @@ def _fetch_config_from_file(config_path: str) -> dict[str, str]:
     """
     try:
         with open(config_path, encoding="utf-8") as config_file:
-            config = yaml.safe_load(config_file)
+            config_input = yaml.safe_load(config_file)
 
         if KedroCliError.VERBOSE_ERROR:
             click.echo(config_path + ":")
-            click.echo(yaml.dump(config, default_flow_style=False))
+            click.echo(yaml.dump(config_input, default_flow_style=False))
     except Exception as exc:
         raise KedroCliError(
             f"Failed to generate project: could not load config at {config_path}."
         ) from exc
 
-    # The return type defined is more specific than the "Any" type config return from yaml.safe_load
-    return config  # type: ignore[no-any-return]
+    config = (
```

Comment:
Why do we introduce this dictionary comprehension here?

Reply:
> Why do we introduce this dictionary comprehension here?

I made this to be consistent with mypy --strict rules: 
yaml.safe_load(config_file) returns object with Any type. From our function we want to return dict[str, str] object. I think a good way to do so - made an explicit conversion, after that we can remove "type ignore"

Reply:
Instead of this, could we add typing `config_input: dict[str,str] = yaml.safe_load(config_file)` on line 638 instead?

Reply:
great, it works, fixed!